### PR TITLE
Clarify how time is set

### PR DESCRIPTION
**Pull request by andreas-junghanns**

### CODE REVIEW

**Comment by andreas-junghanns** [comment]
This addresses one of the points in #1078.

File: docs/4_2_co-simulation_api.adoc [hunk 1]
```diff
@@ -260,6 +260,8 @@ _For example, if the slave does not use the update formula for the <<independent
 
 It depends on the capabilities of the FMU which argument constellations and calling sequences are allowed (see <<fmi-for-co-simulation>>).
 
+Only <<fmi3DoStep>> can change the time inside a Co-Simulation FMU.
```
Comment (chrbertsch) [inline docs/4_2_co-simulation_api.adoc:263]: would change this to something like
`Only <<fmi3DoStep>> can change the time of a Co-Simulation FMU from the outside (while time advances internally during a communication interval).`

Comment (andreas-junghanns) [reply docs/4_2_co-simulation_api.adoc:263]: Agree, much clearer this way.

**Comment by andreas-junghanns** [comment]
Christian: I think you have to approve again after my change

File: docs/4_2_co-simulation_api.adoc [hunk 2]
```diff
@@ -387,7 +389,7 @@ The state machine in <<figure-co-simulation-state-machine>> defines the supporte
 [#figure-co-simulation-state-machine]
 image::images/state-machine-co-simulation.svg[width=80%, align="center"]
 
-In this Co-Simulation interface the following functions must not be called: <<fmi3ActivateModelPartition>>, <<fmi3NewDiscreteStates>>, <<fmi3EnterEventMode>>, <<fmi3SetClock>>, <<fmi3GetClock>>, <<fmi3SetIntervalFraction>>, <<fmi3GetIntervalFraction>>, <<fmi3SetIntervalDecimal>>, <<fmi3GetIntervalDecimal>> including all functions that are specific to Model Exchange.
+In this Co-Simulation interface the following functions must not be called: <<fmi3SetTime>>, <<fmi3ActivateModelPartition>>, <<fmi3NewDiscreteStates>>, <<fmi3SetContinuousStates>>, <<fmi3EnterEventMode>>, <<fmi3SetClock>>, <<fmi3GetClock>>, <<fmi3SetIntervalFraction>>, <<fmi3GetIntervalFraction>>, <<fmi3SetIntervalDecimal>>, <<fmi3GetIntervalDecimal>> including all functions that are specific to Model Exchange.
```
Comment (IZacharias) [inline docs/4_2_co-simulation_api.adoc:392]: fmi3SetClock and fmi3GetClock are allowed in Co-Simulation

Comment (andreas-junghanns) [reply docs/4_2_co-simulation_api.adoc]: Agreed, fixed.

**Comment by andreas-junghanns** [comment]
web design meeting: we should find a central place (the table in 1.2.4?) to explain the different time concepts for all three interfaces. Also mention super-dense time.

**Comment by andreas-junghanns** [comment]
As discussed during the during the design meeting: please find my proposed changes in this PR.

File: docs/2_1_common_api.adoc [hunk 3]
```diff
@@ -580,6 +580,28 @@ For Co-Simulation FMUs, additional functions are defined in <<transfer-of-input-
...
 
+==== Advancing Time [[advancing-time]]
+
+This section highlights the differences of the concept of time (in general the independent variable) for the three different FMI types, ME, CS and SE.
+
+In Model Exchange, time is under the sole control of the importer and its integration algorithm.
+The model itself receives the current time to be used in its computation with <<fmi3SetTime>>.
+In fact, time is not necessarily advancing linearly as solvers might need to find zero-crossing points or the importer uses the Model Exchange FMU to find points of optimal control.
+
+In Co-Simulation, time advances in variable steps negotiated between the co-simulation algorithm of the importer and the FMU.
```
Comment (chrbertsch) [inline docs/2_1_common_api.adoc:591]: "in variable steps" --> "in (possibly) variable steps" as in many cases fixed communication timesteps are used for "naive" co-simulation

File: docs/2_1_common_api.adoc [hunk 4]
```diff
@@ -580,6 +580,28 @@ For Co-Simulation FMUs, additional functions are defined in <<transfer-of-input-
...
+The importer calls <<fmi3DoStep>> with the <<currentCommunicationPoint>> and a target <<communicationStepSize>> (required to be larger than 0.0).
+During this <<fmi3DoStep>>, both importer and FMU might encounter events that will reduce the <<communicationStepSize>> (potentially even down to 0.0).
+With the mechanisms of the callback <<fmi3CallbackIntermediateUpdate>> and/or <<earlyReturn>>, both importer and FMU can determine if such a step-size reduction is required.
+The output argument <<lastSuccessfulTime>> of <<fmi3DoStep>> allows the FMU to signal the master its current internal time.
+
+In Scheduled Execution, time has a more discrete form.
+The scheduler of the importer activates specific tasks according to the time of the importer.
+However, the time itself is not communicated to the FMU nor its partitions.
```
Comment (chrbertsch) [inline docs/2_1_common_api.adoc:599]: "However, the time itself is not communicated to the FMU nor its partitions." ?? We communicate the activation time to the model partition.

File: docs/2_1_common_api.adoc (hunk 3 above)
Comment (andreas-junghanns) [reply docs/2_1_common_api.adoc]: thanks, fixed.

File: docs/2_1_common_api.adoc (hunk 4 above)
Comment (andreas-junghanns) [reply docs/2_1_common_api.adoc]: thanks (I completely missed this feature!!), fixed

**Pull request merged**